### PR TITLE
Implemented Page Object Model

### PR DESCRIPTION
**Pull request by aneghina**

### CODE REVIEW

File: pom/login.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    async goto() {
+        await this.page.goto('https://www.facebook.com');
+    }
+
+    async close() {
+        await this.page.close();
+    }
+
```
Comment (dzarubingrid) [inline pom/login.spec.ts:24]: Unnecessary empty line

File: pom/login.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { type Locator, type Page } from "@playwright/test"
+
+export class LoginPage {
```
Comment (dzarubingrid) [inline pom/login.spec.ts:3]: The class is called LoginPage, so it must be placed in a file called loginPage.ts

File: tests/test.spec.ts [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
-    await page.goto('https://www.facebook.com');
-    await page.getByTestId('royal_email').fill('anastasia.neghina@gmail.com');
-    await page.getByTestId('royal_pass').fill('An2024grid!');
-    await page.getByRole('button', { name: 'Log in' }).click();
-    await expect(page).toHaveURL(/.*failure/);
-    await page.close();
-  });
+  const loginPageTest = new LoginPage(page);
+  await loginPageTest.goto();
+  await loginPageTest.emailField.fill('ag.neghina@gmail.com');
```
Comment (dzarubingrid) [inline tests/test.spec.ts:7]: Let's extract lines 7-9 in a separate function in the LoginPage class. The usage would be await loginPage.login('ag.neghina@gmail.com', 'An2024grid!');

File: tests/test.spec.ts [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+  await loginPageTest.passwordField.fill('An2024grid!');
+  await loginPageTest.loginButton.click();
+  await expect(loginPageTest.loginButton).toBeVisible();
```
Comment (dzarubingrid) [inline tests/test.spec.ts:10]: No bueno. We must validate for sure that we are logged in

File: tests/test.spec.ts [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
+  await loginPageTest.close();
```
Comment (dzarubingrid) [inline tests/test.spec.ts:11]: There is no need to close the page after the test - Playwright will do it

File: tests/test.spec.ts [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
-    await page.getByTestId('royal_email').fill('anastasia.neghina@gmail.com');
-    await page.getByTestId('royal_pass').fill('An2024grid!');
-    await page.getByRole('button', { name: 'Log in' }).click();
-    await expect(page).toHaveURL(/.*failure/);
-    await page.close();
-  });
+  const loginPageTest = new LoginPage(page);
+  await loginPageTest.goto();
+  await loginPageTest.login('ag.neghina@gmail.com', 'An2024grid!');
+  await loginPageTest.loginButton.click();
```
Comment (dzarubingrid) [inline tests/test.spec.ts:8]: Please move this line into loginPage.login(). As the function name implies, user must be logged in when the function is called

File: tests/test.spec.ts [hunk 7]
```diff
@@ -1,15 +1,17 @@
...
 
 test('@functional - check login page negativ test', async ({ page }) => {
-    await page.goto('https://www.facebook.com');
-    await page.getByTestId('royal_email').fill('anastasia.neghina@gmail.com');
-    await page.getByTestId('royal_pass').fill('An2024grid!');
-    await page.getByRole('button', { name: 'Log in' }).click();
-    await expect(page).toHaveURL(/.*failure/);
-    await page.close();
-  });
+  const loginPageTest = new LoginPage(page);
```
Comment (dzarubingrid) [inline tests/test.spec.ts:5]: loginPageTest => loginPage?

File: tests/test.spec.ts [hunk 8]
```diff
@@ -1,15 +1,17 @@
...
+});
 
-test('@visualregression - screenshot comparison test',async ({page}) => {
-  await page.goto('https://www.facebook.com');
-  expect( await page.screenshot()).toMatchSnapshot('facebook.png')
+test('@visualregression - screenshot comparison test', async ({ page }) => {
+  const loginPageTest = new LoginPage(page);
+  await loginPageTest.goto();
+  expect(await page.screenshot()).toMatchSnapshot();
+  await loginPageTest.close();
```
Comment (dzarubingrid) [inline tests/test.spec.ts:16]: There is no need to close pages :)

File: pom/loginPage.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    constructor(page: Page) {
+        this.page = page;
+        this.emailField = page.getByTestId('royal_email');
+        this.passwordField = page.getByTestId('royal_pass');
+        this.loginButton = page.getByRole('button', { name: 'Log in' });
+        this.errorMessage = page.locator(`//div[@class='_9ay7']`);
+    }
+
+    async goto() {
+        await this.page.goto('https://www.facebook.com');
```
Comment (dzarubingrid) [inline pom/loginPage.ts:20]: Please move the url into the playwright config (baseUrl)

File: pom/loginPage.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import { type Locator, type Page } from "@playwright/test"
+
+export class LoginPage {
+
```
Comment (dzarubingrid) [inline pom/loginPage.ts:4]: Unnecessary blank line

**Comment by aneghina** [comment]
In this new commit, I have:
- Deleted empty lines and unnecessary lines of code
- Moved the URL into the Playwright config
- Changed the name of the constant to 'loginPage'
- Added a 'click()' method to the 'login' function

**Pull request merged**